gateway/gateway_test.go

```diff
@@ -2747,3 +2751,127 @@ func waitForBackendConfigInit(gw *Handle) {
 		2*time.Second,
 	).Should(BeTrue())
 }
+
+func TestLeakyUploader(t *testing.T) {
+	// Setup Minio test container
+	pool, err := dockertest.NewPool("")
+	require.NoError(t, err)
+
+	t.Run("invalid JSON payload should be uploaded to Minio", func(t *testing.T) {
+		minioContainer, err := minio.Setup(pool, t)
+		require.NoError(t, err)
+
+		// create test gateway with leaky uploader enabled
+		gw, cleanupFn := createTestGatewayWithLeakyUploader(t, minioContainer.Endpoint, minioContainer.AccessKeyID, minioContainer.AccessKeySecret)
+		t.Cleanup(cleanupFn)
+
+		// Prepare invalid JSON payload
+		invalidPayload := []byte(`{"invalid": "json`)
+
+		// Call internal batch handler
+		jobs, err := gw.extractJobsFromInternalBatchPayload("batch", invalidPayload)
+
+		// Verify error response
+		require.Error(t, err)
+		require.Equal(t, response.InvalidJSON, err.Error())
+		require.Nil(t, jobs)
+
+		// verify file was uploaded to Minio
+		require.Eventually(t, func() bool {
+			contents, err := minioContainer.Contents(context.Background(), "gw-failed-events")
+			require.NoError(t, err)
```

Comment:
No require should be inside `Eventually`. Otherwise it might fail at the first attempt instead of retrying for 2s like you specified below.